src/App.tsx

```diff
@@ -1,61 +1,117 @@
+import React, { useState, ChangeEvent } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+const startingArray: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [sectionValue, setSectionValue] = useState('');
+  const [inputValue, setInputValue] = useState('');
+  const [todoList, setToDoList] = useState<Todo[]>(startingArray);
+
+  const [titleState, setErrorTitleState] = useState(false);
+  const [selectState, setErrorSelectState] = useState(false);
+
+  const handleSelectChange = (event: ChangeEvent<HTMLSelectElement>) => {
+    setSectionValue(event.target.value);
+    setErrorSelectState(false);
+  };
+
+  const resetForm = () => {
+    setInputValue('');
+    setSectionValue('');
+    setErrorTitleState(false);
+    setErrorSelectState(false);
+  };
+
+  const addTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const newId = Math.max(...todoList.map((elem) => elem.id));
+    const userId = +sectionValue;
+
+    const todo: Todo = {
+      id: newId + 1,
+      title: inputValue,
+      completed: false,
+      userId: 0,
+      user: getUserById(userId),
+    };
+
+    if (inputValue.length === 0) {
```

Comment:
```suggestion
    if (!inputValue) {
```
Just check if a value exists, you don't have to check the length

Fix in all places